src/controllers/user.js

```diff
@@ -31,23 +32,21 @@ module.exports = {
         }
       );
 
-      if (!updatedUser) {
-        res
-          .status(422)
-          .json({ message: "The user with the specified id wasn't found" });
-      } else {
-        res.json(updatedUser);
-      }
+      if (!updatedUser)
+        throw new Error("The user with the specified ID wasn't found");
+
+      res.json(updatedUser);
     } catch (err) {
       res.status(422).json({ message: err.message });
     }
   },
 
+  // TODO: make sure to add authorization
   deleteUser: async (req, res) => {
     const { id } = req.params;
     try {
-      const user = await UserModel.findByIdAndRemove(id);
-      res.status(204).json(user);
+      await UserModel.findByIdAndRemove(id);
+      res.end();
```

Comment:
Thanks for editing this, just as I mentioned in the previous review that express sets the status code to 200 by default. So, you still need to set the status code to 204 --> `res.status(204).end();`